File: 프로그래머스/2/42885. 구명보트/구명보트.py
def solution(people, limit):
    answer = 0
    people.sort(reverse=True)
    front = 0
    end = len(people)-1
    
    while front < end :
        
        if people[front] + people[end] > limit :
            print(people[front])
            front+=1
        
        else :
            if people[front] + people[end] <= limit :
                print(people[front])
                front+=1
                print(people[end])
                end -=1
        answer+=1
    if front == end :
        answer+=1
        
    return answer

File: 프로그래머스/2/42885. 구명보트/구명보트.java
import java.util.*;

class Solution {
    public int solution(int[] people, int limit) {
        int answer = 0;
        Arrays.sort(people);
        int sum = 0;
        int left = 0;
        int right = people.length - 1;
        
        while (left  <= right) {
            answer +=1;
            if(people[right]+people[left] <= limit){
                left ++;
            }
            right --;
        }
        return answer;
    }
}